lib/actions/posts.ts

```diff
@@ -117,12 +117,12 @@ export async function deletePost(post: INote) {
       throw new Error("You must be logged in to delete a post");
     }
 
-    if (post.creator.toString() !== sessionUser.id) {
-      throw new Error("You are not authorized to delete this post");
-    }
-
     //delete the post from the database
-    await Note.deleteOne({ _id: post._id });
+    const { deletedCount } = await Note.deleteOne({ _id: post._id });
```

Comment:
Let's rework this to instead use the session data to make sure we only match the creator.

This way the only time we'll find a Note to delete is if the currently logged in user (Which we're getting from the session) is the creator of the note. 

```js
    const { deletedCount } = await Note.deleteOne({
      _id: postID,
      creator: sessionUser.id,
    });
```